executor/write.go

```diff
@@ -915,19 +911,14 @@ func (e *InsertExec) exec(goCtx goctx.Context, rows [][]types.Datum) (Row, error
 			}
 			txn = e.ctx.Txn()
 			rowCount = 0
-			if !sessVars.ImportingData {
-				writeBufs.BufStore = kv.NewBufferStore(txn, kv.TempTxnMemBufCap)
-			}
 		}
 		if len(e.OnDuplicate) == 0 && !e.IgnoreErr {
 			txn.SetOption(kv.PresumeKeyNotExists, nil)
 		}
 		h, err := e.Table.AddRecord(e.ctx, row, false)
 		txn.DelOption(kv.PresumeKeyNotExists)
 		if err == nil {
-			if !sessVars.ImportingData {
-				e.ctx.StmtAddDirtyTableOP(DirtyTableAddRow, e.Table.Meta().ID, h, row)
-			}
+			e.ctx.StmtAddDirtyTableOP(DirtyTableAddRow, e.Table.Meta().ID, h, row)
```

Comment:
I think we should write binlog, whether or not it's importing data.

Reply:
It has some cost when merging dirty table.

Reply:
ok, `begin; stmt1; stmt2 ...` stmt2 can't see stmt1's write when it's importing data.
I'll revert here.